File: 206 Reverse Linked List/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if (head == NULL || head->next == NULL){
            return head;
        }
        ListNode* prevnode = nullptr; 
        ListNode* nextnode;
        ListNode* cur = head;
        while (cur!=NULL){
            nextnode = cur->next;
            cur->next = prevnode;
            prevnode = cur;
            cur = nextnode;
        }
        return prevnode;
    }
};

File: 206 Reverse Linked List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def reverseList(self, head):
        if head is None or head.next is None:
            return head
        prevnode = None
        cur = head
        while cur is not None:
            nextnode = cur.next
            cur.next = prevnode
            prevnode = cur
            cur = nextnode
        return prevnode
        